lib/datadog/tracing/contrib/aws/configuration/settings.rb

```diff
@@ -26,7 +26,14 @@ class Settings < Contrib::Configuration::Settings
               o.lazy
             end
 
-            option :service_name, default: Ext::DEFAULT_PEER_SERVICE_NAME
+            option :service_name do |o|
+              o.default do
+                ENV.fetch(Ext::ENV_SERVICE_NAME) do
+                  Ext::DEFAULT_PEER_SERVICE_NAME
+                end
+              end
```

Comment:
```suggestion
              o.default { ENV.fetch(Ext::ENV_SERVICE_NAME, Ext::DEFAULT_PEER_SERVICE_NAME }
```

This syntax also works.

I'll leave it to you if which one you like more, doesn't matter much to me.